projects/gloo/pkg/plugins/tracing/plugin.go

```diff
@@ -85,61 +92,83 @@ func (p *Plugin) ProcessHcmSettings(cfg *envoyhttp.HttpConnectionManager, hcmSet
 	return nil
 }
 
-func envoyTracingProvider(tracingSettings *tracing.ListenerTracingSettings) *envoy_config_trace_v3.Tracing_Http {
-	if tracingSettings.Provider == nil {
-		return nil
+func (p *Plugin) ProcessEnvoyTracingProvider(snapshot *v1.ApiSnapshot, tracingSettings *tracing.ListenerTracingSettings) (*envoy_config_trace_v3.Tracing_Http, error) {
```

Comment:
nit: private function
```suggestion
func (p *Plugin) processEnvoyTracingProvider(snapshot *v1.ApiSnapshot, tracingSettings *tracing.ListenerTracingSettings) (*envoy_config_trace_v3.Tracing_Http, error) {
```